activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -377,12 +377,20 @@ def rename_column(table_name, column_name, new_column_name) # :nodoc:
       end
 
       def add_index(table_name, column_name, **options) # :nodoc:
+        create_index = build_create_index_definition(table_name, column_name, **options)
+        return unless create_index
+
+        execute(create_index.ddl)
+      end
+
+      def build_create_index_definition(table_name, column_name, **options)
```

Comment:
Should apps consume this in the ExecutionStrategy? If so it should be documented, if not it should be `# :nodoc:`.